### DISCUSSION TITLE
Connect to server via Python or other languages

### DISCUSSION THREAD

**Discussion by ghost** [apohl79/audiogridder#372] [General]
Hi apohl,
is it possible to use audigridder as a compute engine from other programming languages?

Example:
I would like to load a plugin from python, send the preset or parameter settings and then get the computed audio back.

This would be cool in order to apply algorithms for preset programming or automated rendering of presets for previewing

Thanks Phil

**Comment by apohl79** [comment]
Hey Phil,

AG uses a custom TCP protocol (mainly implemented in https://github.com/apohl79/audiogridder/blob/master/Common/Source/Message.hpp). If you reimplement the client side in python, you can implement a python client. That said, the protocol is changing from time to time and especially the audio streaming will be replaced by RTP at some point.

There is no library, that you can bind to python via swig or so. I would assume that it will require a reimplementation, or you could try to put it into a lib first to create python bindings.

Sounds like a nice idea though. ;-)

Best,
Andreas
